### DISCUSSION TITLE
Stable release

### DISCUSSION THREAD

**Discussion by garak** [deployphp/deployer#2483] [General]
I see that [milestone for 7.0](https://github.com/deployphp/deployer/milestone/7) is 100% complete and unchanged in the last couple of weeks.
Is there any issue that is blocking release of 7.0? If so, what about adding it, so it can be followed/helped with/etc.?

**Comment by aanderse** [comment]
Any update on this? I'm interested in using `7.0` in production but would like a stable release.

**Reply by antonmedv** [reply]
Docs for v7 still under development. As soon as docs are ready, will release v7.

**Reply by garak** [reply]
Can I help with that?

**Reply by antonmedv** [reply]
Definitely! I will create issue with todo for docs.

**Reply by jasperf** [reply]
Yeah, I can help out with docs as well. Will be great to have Deployer 7 stable out so I can curl get it as a global package and use it in Laravel projects using simple `composer require deployer/deployer` to get stable being version 7.

**Reply by s0me0ther** [reply]
When will a stable release be published?

Right now there is a not more maintained version 6, and an unstable beta of version 7

:/

**Reply by antonmedv** [reply]
Don’t know.

**Reply by jasperf** [reply]
Now that I am working on upgrading to PHP 8, also to prepare for Laravel 9 I need to use Deployer 7. This as Deployer 6.8 does not work with PHP 8. And yes, Deployer 7 is still officially beta. This complicates things. So at least an update to let's say Deployer 6.9 with support for PHP 8 would be really helpful until Deployer 7 is stable.

**Reply by garak** [reply]
@jasperf what's the problem with being *officially* beta?

**Reply by jasperf** [reply]
That it is not for production @garak . And as you have to you use `dev-master` you are and taking risks on production using beta and are missing the option to easily choose another stable version.

**Reply by antonmedv** [reply]
Releasing v6 with php8 support is complicated. Due to old php7 only dependences.

**Reply by jasperf** [reply]
Well, Laravel 9 release with need for PHP 8 or Deployer 7 come January 2022 https://laravel.com/docs/8.x/releases . So would be great to have version 7 stable before that data. Just so we can do some testing. I am doing some beta testing on staging already. New `.yml` setup and different syntax takes some time, but I would think most is ready besides documentation. Docs need some love still, but that can be helped. I can help out some if need be.

**Reply by garak** [reply]
> That it is not for production @garak . And as you have to you use `dev-master` you are and taking risks on production using beta and are missing the option to easily choose another stable version.

You don't need to use `dev-master`, just use `^7.0@beta`.
And this is not used in production, unless you deploy something *from* production to elsewhere.

**Reply by jasperf** [reply]
> You don't need to use dev-master, just use ^7.0@beta.

Did not think about that @garak . Thanks for that. As for my reference to production . If you have a live website you normally do not add a package that is beta. This as it refers to not being ready for production. That is all.

**Reply by garak** [reply]
As I mentioned, this package would never end to be in production, just like any other tool required in `require-dev` section of your composer configuration.

**Reply by jasperf** [reply]
@garak True, in ways, however, a beta tool used to deploy code to staging or production is a tool used that is not considered as stable.  And as such when things blow up in your face you are unlikely to get any help as it was labeled beta. More help than when it was alpha and less issues for sure, but less help and no guarantees stable offers still. If the package would have been stable already @antonmedv would have already released a standard version 7. Hopefully that will be soon.

**Reply by antonmedv** [reply]
Mostly documentation is missing.

**Reply by jasperf** [reply]
I can help some with documentation . I was working on a `.yml` file at https://github.com/deployphp/deployer/discussions/2585 . Could be used at https://deployer.org/docs/7.x/yaml or at least as a base . The [Config page](https://deployer.org/docs/7.x/config) is also empty still. Not sure what we need put there. But perhaps details added to a yaml file with descriptions. I could make a start. A note on version 6 vs version 7 would be useful as well I guess. Let me know @antonmedv

**Reply by jasperf** [reply]
Added start at https://github.com/deployphp/deployer/pull/2677 @antonmedv . Hope it is useful. Will add more once usefulness has been confirmed and tips on how to add more.
